core/src/main/java/org/infinispan/remoting/transport/MultiTargetRequest.java

```diff
@@ -0,0 +1,164 @@
+package org.infinispan.remoting.transport;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.infinispan.remoting.responses.CacheNotFoundResponse;
+import org.infinispan.remoting.responses.Response;
+import org.infinispan.util.concurrent.TimeoutException;
+
+import net.jcip.annotations.GuardedBy;
+
+/**
+ * Request implementation that waits for responses from multiple target nodes.
+ *
+ * @author Dan Berindei
+ * @since 9.1
+ */
+public class MultiTargetRequest<T> extends AbstractRequest<T> {
+   @GuardedBy("responseCollector")
+   private final Address[] targets;
+   @GuardedBy("responseCollector")
+   private int missingResponses;
+
+   public MultiTargetRequest(ResponseCollector<T> responseCollector, long requestId, RequestRepository repository,
+                             Collection<Address> targets, Address excluded) {
+      super(requestId, responseCollector, repository);
+      this.targets = new Address[targets.size()];
```

Comment:
1. from `toArray()` javadoc
```
The generator function takes an integer, which is the size of the desired array, and produces an array of the desired size.
```
2. this case is small enough for a single line.

... but ok. I won't insist